Full Program/finalFile.py

```diff
@@ -0,0 +1,155 @@
+from bluetooth import *
+from picamera import PiCamera
+from gpiozero import MotionSensor
+import os
+import sys
+import time
+import datetime as dt
+import wiringpi as GPIO
+
+
+# DEFS
+def initGPIO():
+    GPIO.wiringPiSetupGpio()
+    GPIO.pinMode(23, GPIO.INPUT)  # switching mode
+    GPIO.pinMode(24, GPIO.INPUT)  # capture picture from camera
+    GPIO.pinMode(25, GPIO.INPUT)  # exit from camera
+    GPIO.pinMode(18, GPIO.INPUT)  # motion detector input
+    GPIO.pinMode(27, GPIO.OUTPUT)  # LED indicates switching between modes
+    GPIO.pinMode(22, GPIO.OUTPUT)  # LED indicates taking a photo/video
+    GPIO.pullUpDnControl(23, GPIO.PUD_UP)
+    GPIO.pullUpDnControl(24, GPIO.PUD_UP)
+    GPIO.pullUpDnControl(25, GPIO.PUD_UP)
+
+
+def getFileName(type):
+    now = dt.datetime.now()
+    date = str(now.date()) + "_" + str(now.time().replace(microsecond=0))
+    if (type):  # photo
+        filename = "photo_" + date + '.jpg'
+    else:  # video
+        filename = "video_" + date + '.h264'
+    return filename
+
+
+def changeMode(mode):
+    GPIO.digitalWrite(27, GPIO.HIGH)
+    time.sleep(0.3)
+    mode = (mode + 1) % 3
+    if (mode == 0):
+        print("Photo mode\n")
+    elif (mode == 1):
+        print("Video mode\n")
+    elif (mode == 2):
+        print("Motion Detector Video mode\n")
+    GPIO.digitalWrite(27, GPIO.LOW)
+    return mode
+
+
+def capturePicture():
+    GPIO.digitalWrite(22, GPIO.HIGH)
+    os.system('raspistill -o /home/pi/Desktop/Camera/Photos/' + getFileName(True))
+    print("Just took a photo!\n")
+    GPIO.digitalWrite(22, GPIO.LOW)
+
+
+def captureVideo():
+    GPIO.digitalWrite(22, GPIO.HIGH)
+    cam = PiCamera()
+    cam.start_recording('/home/pi/Desktop/Camera/Videos/' + getFileName(False))
+    time.sleep(5)
+    cam.stop_recording()
+    print("Just recorded a video!\n")
+    GPIO.digitalWrite(22, GPIO.LOW)
+
+
+def captureWithPIR():
+    GPIO.digitalWrite(22, GPIO.HIGH)
+    cam = PiCamera()
+    cam.start_recording('/home/pi/Desktop/Camera/Videos/' + getFileName(False))
+    pir.wait_for_no_motion()
+    cam.stop_recording()
+    print("Just recorded a video after motion detected!\n")
+    GPIO.digitalWrite(22, GPIO.LOW)
+
+
+def cleanUp():
+    # output pins set to 0
+    GPIO.digitalWrite(27, GPIO.LOW)
+    GPIO.digitalWrite(22, GPIO.LOW)
+    # output pins set to input
+    GPIO.pinMode(27, GPIO.INPUT)
+    GPIO.pinMode(22, GPIO.INPUT)
+    sys.exit("Good bye! :)")
+    # clean up bluetooth connection
+    client_sock.close()
+    server_sock.close()
+
+
+initGPIO()
+# HELLO MESSAGES
+print("Photo mode\n")
+
+# BLUETOOTH SETUP
+server_sock = BluetoothSocket(RFCOMM)
+server_sock.bind(("", PORT_ANY))
+server_sock.listen(1)
+
+port = server_sock.getsockname()[1]
+
+uuid = "94f39d29-7d6d-437d-973b-fba39e49d4ee"
```

Comment:
I cannot agree with the last one. UUID must not be changeable, 'cause this number must be identical as in mobile app and there UUID is hardcoded (it should be).
Only variable that can be accessible is name of Raspi's Bluetooth in mobile app.